javalib/src/main/scala/java/lang/String.scala

```diff
@@ -993,37 +1054,63 @@ for (cp <- 0 to Character.MAX_CODE_POINT) {
       prep.append(this.substring(startOfSegment, i)).toString
   }
 
-  private def skipCharsWithCombiningClassOtherThanNoneOrAboveForwards(
-      i: Int): Int = {
+  private def skipConditionalCharsForwards(i: Int)(
+      shouldSkip: Int => scala.Boolean): Int = {
     // scalastyle:off return
-    import Character._
-    val len = length()
+    // use offset + count instead of length to handle Strings created using subString method
+    val len = offset + count
```

Comment:
`length()`, and `codePointAt()` should take care of `offset`. We shouldn't have to do this here. If we were manipulating `value` directly, we would have to do this; but when we're using `length()`, `charAt`, etc., we must not take `offset` into account, because those methods already do that. Perhaps `codePointAt` is wrong?

Reply:
I've reverted changes related to `offset` and `count`, I looks that problem with operations on substring  was occouring due to some other bug which was already fixed in a meantime. Thanks you for for spotting this.